### ISSUE TITLE
Assigning futures from coroutine methods produces "assignment-from-no-return"

### ISSUE THREAD

**Issue by matejcik** [PyCQA/pylint#2902]
this is identical to #1715 except it applies to methods in classes

this is fine:
```python
async def xyz():
    pass

x = xyz()
```

this is not fine:
```python
class C:
    async def xyz(self):
        pass

x = C().xyz()
```

**Comment by PCManticore** [comment]
@matejcik What pylint version you have and what exactly is the second example's output look like?

**Comment by matejcik** [comment]
pylint 2.3.1
astroid 2.2.5
Python 3.6.7 (default, Oct 22 2018, 11:32:17) 
[GCC 8.2.0]

the output with no pylintrc is:
************* Module repro
repro.py:1:0: C0111: Missing module docstring (missing-docstring)
repro.py:1:0: C0103: Class name "C" doesn't conform to PascalCase naming style (invalid-name)
repro.py:1:0: C0111: Missing class docstring (missing-docstring)
repro.py:2:4: C0111: Missing method docstring (missing-docstring)
repro.py:1:0: R0903: Too few public methods (1/2) (too-few-public-methods)
**repro.py:5:0: E1111: Assigning result of a function call, where the function has no return (assignment-from-no-return)**
repro.py:5:0: C0103: Constant name "x" doesn't conform to UPPER_CASE naming style (invalid-name)

-------------------------------------
Your code has been rated at -17.50/10
